trio/_core/_exceptions.py

```diff
@@ -47,10 +47,12 @@ class Cancelled(BaseException):
     then this *won't* catch a :exc:`Cancelled` exception.
 
     Attempting to raise :exc:`Cancelled` yourself will cause a
-    :exc:`RuntimeError`. It would not be associated with a cancel scope and thus
-    not be caught by Trio. Use
-    :meth:`cancel_scope.cancel() <trio.CancelScope.cancel>`
-    instead.
+    :exc:`RuntimeError`. Unless raised within the context of a
+    cancelled cancel scope, it would not be caught by Trio and would
+    crash your program. Use :meth:`cancel_scope.cancel()
+    <trio.CancelScope.cancel>` instead, so Trio can maintain the
+    invariant that a :exc:`Cancelled` exception only gets raised when
+    there's a cancelled cancel scope that's intended to catch it.
```

Comment:
This is probably more details on the internals than we really want people thinking about. How about simplifying down to just: "You cannot raise Cancelled yourself. Attempting to do so will produce a RuntimeError. Use cancel_scope.cancel instead."